src/helpers/images/image_provider.ts

```diff
@@ -0,0 +1,59 @@
+import { FigureSize, Getters } from "../../types";
+import { FileUploaderInterface, ImageProviderInterface } from "../../types/files";
+
+export class ImageProvider implements ImageProviderInterface {
+  private fileUploader: FileUploaderInterface;
+  private maxWidth: number;
+  private maxHeight: number;
+
+  constructor(fileUploader: FileUploaderInterface, getter: Getters) {
+    this.fileUploader = fileUploader;
+    const dim = getter.getSheetViewDimension();
+    this.maxWidth = dim.width;
+    this.maxHeight = dim.height;
```

Comment:
1. It's not the responsibility of the ImageProvider to handle max image size.
it shouldn't even know anything about a spreadsheet.
Its job is to provide an image. Nothing more. 1 class = 1 responsibility

2. the sheet view dimension can change (the user can resize the browser window) => it's wrong to store them at the begining.